models/pull.go

```diff
@@ -636,8 +637,8 @@ func newPullRequestAttempt(repo *Repository, pull *Issue, labelIDs []int64, uuid
 	pr.Index = pull.Index
 	pr.BaseRepo = repo
 	pr.Status = PullRequestStatusChecking
-	if len(patch) > 0 {
-		if err = repo.savePatch(sess, pr.Index, patch); err != nil {
+	if patchFileSize > 0 {
+		if err = repo.savePatch(sess, pr.Index, patchFileName); err != nil {
```

Comment:
I think the patch should be saved even if it's got 0 bytes. The reason is that a previous PR creation attempt (from a different PR, _with_ patch data) could have failed between this step and the commit, so a left-over patch file (named after this same index number) might still be lingering in the directory. I hope that makes sense.

In fact, there are scenarios where that patch could be broken anyway with this retry mechanism. Mmm.... we should be locking that path, at the very least? Sorry, I wasn't aware that there were destinations other than the database when I wrote the retry code.

Reply:
It would make it simpler if we dropped that file size test.

Tbh I'm not certain that the patch files are ever actually cleaned up. I don't remember seeing any code looking at deleting them. I actually forgot to look explicitly - that's not like me!!

You're also right about the lack of file locking. This might be because we have generally migrated from the repoworkingcopy lock mechanism to rely on git locking but obviously this doesn't use that mechanism. #9302 obviously isn't affected by this.

I guess the question is how important is it provide a backport?

If we're not going to backport - then I'm happy to close this and move to the more radical refactor. 

(It's not like this is the only place we still read in potentially huge amounts of data - diff and blame are both still doing that, and a change to streaming there will require huge refactoring.)

Reply:
Well, backporting is not my call, but I'm OK with not backporting this as long as #9302 makes it in time for 1.11.